### PR TITLE
Remove ES_NEXT macro

### PR DESCRIPTION
**Pull request by szilagyiadam**
- remove all '#JERRY_ESNEXT' macro
- remove 5.1 build profile, update test runner accordingly (Note: all builtins are turn on by default)
- move tests from tests/jerry/esnext into tests/jerry, concatenate files with same names
- add skiplist to some snapshot tests that were supported only in 5.1
- fix doxygen issues that were hiden before (bc. of es.next macro)

Co-authored-by: Martin Negyokru negyokru@inf.u-szeged.hu
JerryScript-DCO-1.0-Signed-off-by: Adam Szilagyi aszilagy@inf.u-szeged.hu

### CODE REVIEW

**Comment by lygstate** [comment]
neat, May separate the files rename to separate PR?

File: jerry-core/parser/js/common.c [hunk 1]
```diff
@@ -396,7 +394,6 @@ util_print_cbc (ecma_compiled_code_t *compiled_code_p) /**< compiled code */
   JERRY_DEBUG_MSG ("  Const literal range end: %d\n", (int) const_literal_end);
   JERRY_DEBUG_MSG ("  Literal range end: %d\n\n", (int) literal_end);
 
-#if JERRY_ESNEXT || JERRY_FUNCTION_TO_STRING
```
Comment (rerobika) [inline jerry-core/parser/js/common.c:399]: keep JERRY_FUNCTION_TO_STRING

File: jerry-core/parser/js/js-parser.c [hunk 2]
```diff
@@ -907,24 +900,17 @@ parser_post_processing (parser_context_t *context_p) /**< context */
   {
     total_size += sizeof (ecma_value_t);
   }
-#endif /* JERRY_ESNEXT */
 
 #if JERRY_LINE_INFO
   total_size += sizeof (ecma_value_t);
 #endif /* JERRY_LINE_INFO */
-
-#if JERRY_ESNEXT || JERRY_FUNCTION_TO_STRING
```
Comment (rerobika) [inline jerry-core/parser/js/js-parser.c:916]: keep JERRY_FUNCTION_TO_STRING

File: jerry-core/parser/js/js-parser.c [hunk 3]
```diff
@@ -946,16 +932,12 @@ parser_post_processing (parser_context_t *context_p) /**< context */
     total_size += ecma_extended_info_get_encoded_length ((uint32_t) (context_p->function_end_p - function_start_p));
   }
 #endif /* JERRY_FUNCTION_TO_STRING */
-
-#if JERRY_ESNEXT || JERRY_FUNCTION_TO_STRING
```
Comment (rerobika) [inline jerry-core/parser/js/js-parser.c:950]: ditto

File: jerry-core/parser/js/js-parser.c [hunk 4]
```diff
@@ -1356,13 +1331,11 @@ parser_post_processing (parser_context_t *context_p) /**< context */
     compiled_code_p->status_flags |= CBC_CODE_FLAGS_HAS_TAGGED_LITERALS;
     *(--base_p) = (ecma_value_t) context_p->tagged_template_literal_cp;
   }
-#endif /* JERRY_ESNEXT */
 
 #if JERRY_LINE_INFO
   ECMA_SET_INTERNAL_VALUE_POINTER (base_p[-1], line_info_p);
 #endif /* JERRY_LINE_INFO */
 
-#if JERRY_ESNEXT || JERRY_FUNCTION_TO_STRING
```
Comment (rerobika) [inline jerry-core/parser/js/js-parser.c:1365]: keep JERRY_FUNCTION_TO_STRING

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/base/ecma-error-messages.inc.h [hunk 5]
```diff
@@ -250,21 +225,13 @@ ECMA_ERROR_DEF (ECMA_ERR_DATA_VIEW_NOT_SUPPORTED, "DataView support is disabled"
...
 #endif /* JERRY_BUILTIN_DATE */
-#if JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_LOCAL_VARIABLE_IS_REDECLARED, "Local variable is redeclared")
-#endif /* JERRY_ESNEXT */
 ECMA_ERROR_DEF (ECMA_ERR_UNSUPPORTED_BINARY_OPERATION, "Unsupported binary operation")
 #if JERRY_BUILTIN_REGEXP
 ECMA_ERROR_DEF (ECMA_ERR_UNTERMINATED_CHARACTER_CLASS, "Unterminated character class")
 #endif /* JERRY_BUILTIN_REGEXP */
-#if JERRY_BUILTIN_DATAVIEW || JERRY_BUILTIN_TYPEDARRAY || JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_ARRAYBUFFER_IS_DETACHED, "ArrayBuffer has been detached")
```
Comment (lygstate) [inline jerry-core/ecma/base/ecma-error-messages.inc.h:233]: Does JERRY_ESNEXT implies ECMA_ERR_ARRAYBUFFER_IS_DETACHED?

File: jerry-core/ecma/base/ecma-error-messages.inc.h [hunk 6]
```diff
@@ -430,39 +371,25 @@ ECMA_ERROR_DEF (ECMA_ERR_INVALID_SCOPE_CHAIN_INDEX_FOR_EVAL, "Invalid scope chai
 #if JERRY_BUILTIN_TYPEDARRAY
 ECMA_ERROR_DEF (ECMA_ERR_MAXIMUM_TYPEDARRAY_SIZE_IS_REACHED, "Maximum TypedArray size is reached")
 #endif /* JERRY_BUILTIN_TYPEDARRAY */
-#if JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_THE_GIVEN_ARGUMENT_IS_NOT_A_SYMBOL, "The given argument is not a Symbol")
 ECMA_ERROR_DEF (ECMA_ERR_PARAMETER_REJECT_MUST_BE_CALLABLE, "'reject' parameter must be callable")
-#endif /* JERRY_ESNEXT */
-#if JERRY_BUILTIN_TYPEDARRAY || JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_ARGUMENT_THIS_NOT_TYPED_ARRAY, "Argument 'this' is not a TypedArray")
-#endif /* JERRY_BUILTIN_TYPEDARRAY \
```
Comment (lygstate) [inline jerry-core/ecma/base/ecma-error-messages.inc.h:439]: Does this means JERRY_BUILTIN_TYPEDARRAY are always enabled by JERRY_ESNEXT

File: jerry-core/ecma/base/ecma-error-messages.inc.h [hunk 7]
```diff
@@ -430,39 +371,25 @@ ECMA_ERROR_DEF (ECMA_ERR_INVALID_SCOPE_CHAIN_INDEX_FOR_EVAL, "Invalid scope chai
...
 ECMA_ERROR_DEF (ECMA_ERR_SUPER_BINDING_MUST_BE_A_CONSTRUCTOR, "Super binding must be a constructor")
-#endif /* JERRY_ESNEXT */
 #if JERRY_BUILTIN_BIGINT
 ECMA_ERROR_DEF (ECMA_ERR_VALUE_CANNOT_BE_CONVERTED_TO_BIGINT, "Value cannot be converted to BigInt")
 #endif /* JERRY_BUILTIN_BIGINT */
-#if JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_PARAMETER_RESOLVE_MUST_BE_CALLABLE, "'resolve' parameter must be callable")
-#endif /* JERRY_ESNEXT */
-#if JERRY_BUILTIN_DATE || JERRY_ESNEXT
 ECMA_ERROR_DEF (ECMA_ERR_ARGUMENT_THIS_NOT_DATE_OBJECT, "Argument 'this' is not a Date object")
```
Comment (lygstate) [inline jerry-core/ecma/base/ecma-error-messages.inc.h:392]: Does this means JERRY_BUILTIN_DATE always enabled?

File: jerry-core/ecma/base/ecma-error-messages.inc.h [hunk 8]
```diff
@@ -476,19 +403,16 @@ ECMA_ERROR_DEF (ECMA_ERR_CONSTRUCTOR_INT8_ARRAY_REQUIRES_NEW, "Constructor Int8A
 #if JERRY_BUILTIN_CONTAINER
 ECMA_ERROR_DEF (ECMA_ERR_CONTAINER_IS_NOT_A_CONTAINER_OBJECT, "Container is not a container object.")
 #endif /* JERRY_BUILTIN_CONTAINER */
-#if JERRY_BUILTIN_DATE && JERRY_ESNEXT
+#if JERRY_BUILTIN_DATE
 ECMA_ERROR_DEF (ECMA_ERR_INVALID_ARGUMENT_TYPE_IN_TOPRIMITIVE, "Invalid argument type in toPrimitive")
```
Comment (lygstate) [inline jerry-core/ecma/base/ecma-error-messages.inc.h:407]: And this is different from 
```
ECMA_ERROR_DEF (ECMA_ERR_ARGUMENT_THIS_NOT_DATE_OBJECT, "Argument 'this' is not a Date object")
```

**Review by rerobika — APPROVED** [review]

**Pull request merged**